pkg/resources/pods.go

```diff
@@ -99,33 +99,70 @@ func (pa PodAccessor) NotReadyCount() (int, error) {
 	return nr, err
 }
 
-// PodIPsByAge returns the list of running pod (terminating
-// and non-running are excluded) IP addresses, sorted descending by pod age.
-func (pa PodAccessor) PodIPsByAge() ([]string, error) {
+// PodFilter provides a way to filter pods for a revision.
+// Returning true, means that pod should be kept.
+type PodFilter func(p *corev1.Pod) bool
+
+// PodTransformer provides a way to do something with the pod
+// that has been selected by the filters.
+// For example pod transformer may extract a field and store it in
+// internal state.
+type PodTransformer func(p *corev1.Pod)
+
+// ProcessProd filters all the pods using provided pod filters and then if the pod
```

Comment:
```suggestion
// ProcessPods filters all the pods using provided pod filters and then if the pod
```

Reply:
nit: maybe high light that a pod is selected if all filters return true?

Reply:
Freudian typo.

Reply:
Sure added `all`.